src/bot/mtproto-api/bot/session/session-manager.ts

```diff
@@ -0,0 +1,38 @@
+import { Context } from "../../../../types";
+import { GithubStorage } from "../../../../adapters/github/storage-layer";
+import { SuperbaseStorage } from "../../../../adapters/supabase/supabase";
+import { GitHubSession } from "./github-session";
+import { SupabaseSession } from "./supabase-session";
+import { StringSession } from "telegram/sessions";
+
+export interface SessionManager extends StringSession {
+  storage: GithubStorage | SuperbaseStorage;
+  context: Context;
+  session?: string;
+  saveSession(): Promise<void>;
+  loadSession(): Promise<GitHubSession | SupabaseSession>;
+  getSession(): Promise<string>;
+  deleteSession(): Promise<void>;
+}
+
+export class SessionManagerFactory {
+  // eslint-disable-next-line sonarjs/public-static-readonly
+  static sessionManager: SessionManager;
+  // eslint-disable-next-line sonarjs/public-static-readonly
```

Comment:
Why did you break this rule?

Reply:
can't assign to `readonly` and I assign to storage just below

```ts
this.storage = this.sessionManager.storage;
```